### ISSUE TITLE
Something went wrong If this error persists, please report an issue here in Model section

### ISSUE THREAD

**Issue by Gangadharsmg** [mlflow/mlflow#3952]
When I send a ModelSignature while registering a model, if I check it in mlflow UI then i'm getting "Something went wrong If this error persists, please report an issue here" in Model section. Please someone can help me.

**Comment by harupy** [comment]
Could you check what the dev console on your browser say?

**Comment by Gangadharsmg** [comment]
> Could you check what the dev console on your browser say?

Thanks for replying.
In console, it's not reporting any errors but it's showing the above message in browser MLFlow server UI.

**Comment by harupy** [comment]
Strange. Do you have code to reproduce the error?

**Comment by Gangadharsmg** [comment]
> Strange. Do you have code to reproduce the error?

Yes, actually I'm using the sklearn_elastic_wine project which is in mlflow/examples folder. But for the same code i'm trying to use ModelSignature part.
Below code is what I'm trying. It registering a model in model registry. But if I try to open this particular version, above message will come.

`import logging
import sys
from urllib.parse import urlparse

import mlflow.sklearn
import numpy as np
import pandas as pd
import warnings
from mlflow.models.signature import ModelSignature
from mlflow.models.signature import infer_signature
from sklearn.linear_model import ElasticNet
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
from sklearn.model_selection import train_test_split

logging.basicConfig(level=logging.WARN)
logger = logging.getLogger(__name__)


def eval_metrics(actual, pred):
    rmse = np.sqrt(mean_squared_error(actual, pred))
    mae = mean_absolute_error(actual, pred)
    r2 = r2_score(actual, pred)
    return rmse, mae, r2


if __name__ == "__main__":
    warnings.filterwarnings("ignore")
    np.random.seed(40)

    # Read the wine-quality csv file from the URL
    csv_url = (
        "http://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv"
    )
    try:
        data = pd.read_csv(csv_url, sep=";")
    except Exception as e:
        logger.exception(
            "Unable to download training & test CSV, check your internet connection. Error: %s", e
        )

    # Split the data into training and test sets. (0.75, 0.25) split.
    train, test = train_test_split(data)

    # The predicted column is "quality" which is a scalar from [3, 9]
    train_x = train.drop(["quality"], axis=1)
    test_x = test.drop(["quality"], axis=1)
    train_y = train[["quality"]]
    test_y = test[["quality"]]

    alpha = float(sys.argv[1]) if len(sys.argv) > 1 else 0.5
    l1_ratio = float(sys.argv[2]) if len(sys.argv) > 2 else 0.5

    with mlflow.start_run():
        lr = ElasticNet(alpha=alpha, l1_ratio=l1_ratio, random_state=42)
        lr.fit(train_x, train_y)

        predicted_qualities = lr.predict(test_x)

        (rmse, mae, r2) = eval_metrics(test_y, predicted_qualities)

        print("Elasticnet model (alpha=%f, l1_ratio=%f):" % (alpha, l1_ratio))
        print("  RMSE: %s" % rmse)
        print("  MAE: %s" % mae)
        print("  R2: %s" % r2)

        mlflow.log_param("alpha", alpha)
        mlflow.log_param("l1_ratio", l1_ratio)
        mlflow.log_metric("rmse", rmse)
        mlflow.log_metric("r2", r2)
        mlflow.log_metric("mae", mae)

        tracking_url_type_store = urlparse(mlflow.get_tracking_uri()).scheme

        # Model registry does not work with file store
        if tracking_url_type_store != "file":
            #
            #         # Register the model
            #         # There are other ways to use the Model Registry, which depends on the use case,
            #         # please refer to the doc for more information:
            #         # https://mlflow.org/docs/latest/model-registry.html#api-workflow
            # Implementing model signature part
            signature = infer_signature(train_x, lr.predict(train_x))
            mlflow.sklearn.log_model(lr, "model", registered_model_name="ElasticnetWineModel", signature=signature)
        else:
            mlflow.sklearn.log_model(lr, "model")`

**Comment by harupy** [comment]
Thanks for the code. What is your mlflow version?

**Comment by Gangadharsmg** [comment]
> @Gangadharsmg
> 
> Thanks for the code. What is your mlflow version?
> 
> I wasn't able to repro the errorr in the latest version of mlflow.

@harupy My MLFlow version is 1.12.1

**Comment by harupy** [comment]
I was able to repro the issue in MLflow 1.12.1.

**Comment by Gangadharsmg** [comment]
> I was able to repro the issue in MLflow 1.12.1. The error originates from line breaks (inserted by `pyyaml`). mlflow 1.12.1 uses `yamljs` which fails to parse a wrapped string field. mlflow >= 1.13 uses `js-yaml` to address this issue.
> 
> ```yaml
> artifact_path: model
> flavors:
>   python_function:
>     env: conda.yaml
>     loader_module: mlflow.sklearn
>     model_path: model.pkl
>     python_version: 3.6.10
>   sklearn:
>     pickled_model: model.pkl
>     serialization_format: cloudpickle
>     sklearn_version: 0.24.0
> run_id: 1aef3874d4424375a6be6f135a41206a
> signature:
>   inputs: '[{"name": "fixed acidity", "type": "double"}, {"name": "volatile acidity",  <- line break 🔥
>     "type": "double"}, {"name": "citric acid", "type": "double"}, {"name": "residual
>     sugar", "type": "double"}, {"name": "chlorides", "type": "double"}, {"name": "free
>     sulfur dioxide", "type": "double"}, {"name": "total sulfur dioxide", "type": "double"},
>     {"name": "density", "type": "double"}, {"name": "pH", "type": "double"}, {"name":
>     "sulphates", "type": "double"}, {"name": "alcohol", "type": "double"}]'
>   outputs: '[{"type": "double"}]'
> utc_time_created: '2021-01-07 10:42:41.752492'
> ```
> 
> Could you try mlflow 1.13.1?

@harupy Thanks for your suggestion.
But after updating to 1.13.1 version also, I'm getting the same issue.

**Comment by harupy** [comment]
@Gangadharsmg

Does the error still persist if you uninstall 1.12.1, then install MLflow 1.13.1?

```python
pip uninstall mlflow
python -c "import mlflow # make sure mlflow uninstalled"
pip install mlflow 
```

**Comment by snewhouse** [comment]
Hi all 
I seem to be getting the same error with `mlflow, version 1.14.0`
